src/riak_kv_vnode.erl

```diff
@@ -1476,147 +1478,169 @@ prepare_put(State=#state{vnodeid=VId,
                              lww=LWW,
                              coord=Coord,
                              robj=RObj,
-                             starttime=StartTime}) ->
+                             starttime=StartTime,
+                             bprops = BProps}) ->
     %% Can we avoid reading the existing object? If this is not an
     %% index backend, and the bucket is set to last-write-wins, then
     %% no need to incur additional get. Otherwise, we need to read the
     %% old object to know how the indexes have changed.
+    IndexBackend = is_indexed_backend(Mod, Bucket, ModState),
+    IsSearchable = ?IS_SEARCH_ENABLED_FOR_BUCKET(BProps),
+    RequiresReadBeforeWrite = LWW andalso (not IndexBackend) andalso (not IsSearchable),
```

Comment:
It seems like RequiresReadBeforeWrite is named correctly. It should either be renamed to something like BlindPut or the test and following case statement should be negated.